src/core/packet.lua

```diff
@@ -75,15 +84,49 @@ local function freelist_nfree(freelist)
    return freelist.nfree
 end
 
-local max_packets = 1e6
+local function freelist_lock(freelist)
+   sync.lock(freelist.lock)
+end
+
+local function freelist_unlock(freelist)
+   sync.unlock(freelist.lock)
+end
+
 local packet_allocation_step = 1000
 local packets_allocated = 0
-local packets_fl = ffi.new("struct freelist", max_packets, 0, max_packets)
+local packets_fl = ffi.new("struct freelist")
+      packets_fl.max = max_packets
```

Comment:
probably should indent this assignment flush left

Reply:
I believe you might even be able to pass `{ max = max_packets }` as the initarg!